client/packages/common/src/ui/forms/JsonForms/components/DateTime.tsx

```diff
@@ -63,22 +97,15 @@ const UIComponent = (props: ControlProps) => {
         <FormLabel sx={{ fontWeight: 'bold' }}>{label}:</FormLabel>
       </Box>
       <Box flexBasis={FORM_INPUT_COLUMN_WIDTH}>
-        <BaseDateTimePickerInput
-          // undefined is displayed as "now" and null as unset
-          value={data ?? null}
-          onChange={e => {
-            try {
-              setError('');
-              if (e) handleChange(path, e.toISOString());
-            } catch (err) {
-              setError((err as Error).message);
-              console.error(err);
-            }
-          }}
-          inputFormat="dd/MM/yyyy hh:mm"
-          error={error || props.errors}
-          // readOnly={!!props.uischema.options?.['readonly']}
-        />
+        {!dateOnly ? (
+          <BaseDateTimePickerInput
+            // undefined is displayed as "now" and null as unset
+            {...sharedComponentProps}
+            error={error || props.errors}
```

Comment:
Is this dateOnly? or readOnly? There could still be user input errors when dateOnly can't there? Is it a problem to always set the error prop even if there can't be an error? (theoretically there could be an error if the initial data is corrupt) then you don't have to handle the cases differently...

Reply:
I haven't changed this prop at all in this PR, so haven't really given too much thought to its correctness.